<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!-- Gazebo world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find challenge2021)/world/challenge_world.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>
  
  <!-- World control -->
  <node pkg="challenge2021" name="World_control" type="world_control.py" output="screen" required="true"/>

  <!-- Spawn turtlebot in Gazebo -->
  <param name="robot_description" command="$(find xacro)/xacro $(find turtlebot3_description)/urdf/turtlebot3_burger_for_autorace_2020.urdf.xacro " />
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0.0 -y 0.0 -z 0.0 -Y 1.557 -param robot_description" />

  <!-- RVIZ -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  <!--teleoperation-->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find challenge2021)/rviz/config.rviz"/>

  <node pkg="mybot_teleop" name="teleop" type="mybot_teleop.py"
          output="screen" required="true"/>
  <param name="linear_scale" value="0.20" type="double"/>
  <param name="angular_scale" value="0.5" type="double"/>
  <param name="which_topic" value="/cmd_vel" type="string"/>

	  
</launch>
